docs/design/architectural-overview-v1.md

```diff
@@ -0,0 +1,105 @@
+
+#### [Limitador Operator:](https://github.com/Kuadrant/limitador-operator)
+* Installs and configures the Limitador data plane component. Reconciles the limits CR into configuration mounted into limitador.
+
+#### [Authorino Operator:](https://github.com/Kuadrant/authorino-operator)
+* Installs and configures the Authorino data plane component.
+
+#### [Cert-Manager:](https://cert-manager.io/)
+* Manages TLS certificates for our components and for the Gateways. Consumes Certificate resources created by Kuadrant operator in response to the TLSPolicy.
+
+#### [DNS Operator](https://github.com/Kuadrant/dns-operator)
+* DNS operator consumes DNSRecord resources that are configured via the DNSPolicy api and applies them into the targeted cloud DNS provider
+AWS, Azure and Google DNS are our main targets
+
+### Data Plane Components and Responsibilities
+
+![](./images/data-plane-overview.jpg)
+
+The data plane components sit in the request flow and are responsible for enforcing configuration defined by policy and providing service protection capabilities based on configuration managed and created by the control plane.
+
+#### [Limitador](https://github.com/Kuadrant/limitador)
+* Complies with the with Envoy rate limiting API to provide rate limiting to the gateway. Consumes limits from a configmap created based on the RateLimitPolicy API.
+
+#### [Authorino](https://github.com/Kuadrant/authorino)
+* Complies with the Envoy external auth API to provide auth integration to the gateway. It provides both Authn and Authz. Consumes AuthConfigs created by the kuadrant operator based on the defined `AuthPolicy` API.
+
+#### [WASM Shim](https://github.com/Kuadrant/wasm-shim)
+* Uses the [Proxy WASM ABI Spec](https://github.com/proxy-wasm/spec) to integrate with Envoy and provide filtering and connectivity to Limitador for request time enforcement of and rate limiting.
+
+
+### Single Cluster Layout 
+
+In a single cluster, you have the Kuadrant control plane and data plane sitting together. It is configured to integrate with Gateways on the same cluster and configure a DNS zone via a DNS provider secret (configured alongside a DNSPolicy). Storage of rate limit counters is possible but not required as they are not being shared.
+
+![](./images/single-cluster-layout.jpg)
+
+
+### Multi-Cluster 
+
+In the default multi-cluster setup. Each individual cluster has Kuadrant installed. Each of these clusters are unaware of the other. They are effectively operating as single clusters. The multi-cluster aspect is created by sharing access with the DNS zone, using a shared host across the clusters and leveraging shared counter storage. 
+The zone is operated on independently by the DNS operator on each cluster to form a single cohesive record set. More details on this can be found in the following RFC document: TODO add link.
```

Comment:
```suggestion

The zone is operated on independently by the DNS operator on each cluster to form a single cohesive record set. More details on this can be found in the following RFC document: TODO add link.
```